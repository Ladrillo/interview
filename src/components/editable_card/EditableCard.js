import React, { Component } from 'react';
import StyledButton from '../../styled/button';
import './EditableCard.css';


export default class EditableCard extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name:     this.props.name,
            birthday: this.props.birthday,
        };

        this.save           = this.save.bind(this);
        this.cancel         = this.cancel.bind(this);
        this.changeName     = this.changeName.bind(this);
        this.changeBirthday = this.changeBirthday.bind(this);
    }

    save(e) {
        const { name, birthday } = this.state;
        const { id, onSave }     = this.props;

        if (! name || ! birthday) {
            alert('missing required fields');
            return;
        }

        onSave({ id, name, birthday });
    }

    cancel(e) {
         this.props.onCancel(this.props.id);
    }

    changeName(e) {
        this.setState({ name: e.target.value });
    }

    changeBirthday(e) {
        this.setState({ birthday: e.target.value });
    }

    render() {
        const { changeName, changeBirthday, save, cancel } = this;
        const { image, homeworld }                         = this.props;
        const { name, birthday }                           = this.state;

        return (
            <div className='card'>
                <div className='card-content'>
                    <div className='card-name'>
                        <input
                            value     = {name}
                            className = 'name'
                            onChange  = {changeName}
                        />
                    </div>
                    <img src={`http://localhost:3008/${image}`} alt='profile' />
                    <p>
                        <span>Birthday:</span>
                        <input
                            value     = {birthday}
                            className = 'birthday'
                            onChange  = {changeBirthday}
                        />
                    </p>
                    <p>
                        <span>Homeworld:</span>
                        <span>{homeworld}</span>
                    </p>
                </div>
                <div className='card-buttons'>
                    <StyledButton className='save-button' onClick={save}>save</StyledButton>
                    <StyledButton className='cancel-button' onClick={cancel}>cancel</StyledButton>
                </div>
            </div>
        );
    }
}