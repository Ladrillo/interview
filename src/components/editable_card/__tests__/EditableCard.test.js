import EditableCard from '../EditableCard';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


describe('EditableCard', () => {

    it('renders correctly', () => {
        const rendered = renderer.create(
            <EditableCard
                key           = {1}
                id            = {1}
                name          = {'Luke Skywalker'}
                image         = {'luke_skywalker.jpg'}
                birthday      = {'XXXOOO'}
                homeworldName = {'Tatooine'}
                homeworldId   = {1}
                onSave        = {() => ({})}
                onCancel      = {() => ({})}
                planets       = {[]}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
