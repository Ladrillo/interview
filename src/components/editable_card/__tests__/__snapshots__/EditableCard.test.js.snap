exports[`EditableCard renders correctly 1`] = `
<div
  className="card">
  <div
    className="card-content">
    <div
      className="card-name">
      <input
        className="name"
        name="name"
        onChange={[Function]}
        value="Luke Skywalker" />
    </div>
    <img
      alt="profile"
      src="http://localhost:3008/luke_skywalker.jpg" />
    <p>
      <span>
        Birthday:
      </span>
      <input
        className="birthday"
        name="birthday"
        onChange={[Function]}
        value="XXXOOO" />
    </p>
    <p>
      <span>
        Homeworld:
      </span>
      <span>
        <select
          className="homeworld"
          name="homeworld"
          onChange={[Function]}
          value={1} />
      </span>
    </p>
  </div>
  <div
    className="card-buttons">
    <div
      className="save-button sc-bdVaJa Tteup"
      onClick={[Function]}>
      save
    </div>
    <div
      className="cancel-button sc-bdVaJa Tteup"
      onClick={[Function]}>
      cancel
    </div>
  </div>
</div>
`;
